### DISCUSSION TITLE
gatsby-source-shopify plugin update to support Shopify Admin GQL API version 2024-04 and beyond?

### DISCUSSION THREAD

**Discussion by JWizerd** [gatsbyjs/gatsby#39129] [Help] [answered]
Hello! I was wondering if there is any plans to create a new version of the **[gatsby-source-shopify](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-shopify)** which will support newer versions of the Shopify Admin GQL API? 

Upon investigation, I noticed that version 2024-04 and beyond is not supported by the plugin as there were breaking changes between the Shopify Admin GQL schema in version 2024-01 and 2024-04. The queries are written to support version 2024-01 and older. 

For example, when attempting to build using 2024-04 the build fails due that an error occurring within the [product query](https://github.com/search?q=repo%3Agatsbyjs%2Fgatsby%20%2Fgatsby-source-shopify%20mediaCount&type=code): 

`Error: query: Invalid bulk query: Field must have selections (field 'mediaCount' re
  turns Count but has no selections. Did you mean 'mediaCount { ... }'?)`

This is but one of many fields that has been updated in the Shopify Admin GQL API between 2024-01 and 2024-04.

This is a pretty major problem for merchants using Shopify since the latest supported version of the plugin is **2024-01** which will sunset in January 2025 as part of [Shopify's API versioning policy ](https://shopify.dev/docs/api/usage/versioning).

**Comment by serhalp** [comment, marked as answer]
Hi @JWizerd. We're actually actively working on this and hope to have a release in the next few weeks.

Would you be in interested in beta testing this release when it's ready?

cc @pieh

**Reply by Licceeee** [reply]
Hi @serhalp, has it been already released?

**Reply by serhalp** [reply]
Yes, it was released in gatsby-source-shopify@9.0.0: https://github.com/gatsbyjs/gatsby/releases/tag/gatsby-source-shopify%409.0.0

**Reply by slaterslater** [reply]
is there a version of **gatsby-source-shopify** that will support Shopify Admin GQL 2024-07 and beyond?

**Reply by serhalp** [reply]
Not yet, but one will be released before the End of Life of 2024-04 (i.e. within the next 20 days).

**Reply by pieh** [reply]
We updated `gatsby-source-shopify` to use Shopify's 2025-01 API version and released `gatsby-source-shopify@10.0.0` today. 

You can learn more about breaking changes (dictated by Shopify API removing some fields) and deprecations in description of https://github.com/gatsbyjs/gatsby/pull/39247

**Comment by pieh** [comment]
There is canary available using 2024-04 shopify API: `gatsby-source-shopify@9.0.0-canary.0`. 

https://github.com/gatsbyjs/gatsby/pull/39082 have changes being made - it should not result in breaking changes for gatsby queries - data types and shapes should be preserved

**Reply by Mandaline** [reply]
Hi, @pieh @serhalp we're on version 2023-07 of the API and getting the same error mentioned above:
Error: query: Invalid bulk query: Field must have selections (field 'mediaCount' returns Count but has no selections. Did you mean 'mediaCount { ... }'?)
Any idea what could be happening?

**Reply by serhalp** [reply]
@Mandaline What do you mean by "we're on version 2023-07"? The latest gatsby-source-shopify uses version 2024-04.

**Reply by Mandaline** [reply]
@serhalp we're on gatsby-source-shopify v5.7.0, as it's an app that we're no longer developing, only providing occasional support. The client doesn't have much budget for upgrades, so we were trying to pinpoint why we were getting that error. The OP was getting the same error but using a newer version of the Shopify API, so I was curious if was in fact breaking changes or if it was coming from something else. I suppose we will need to upgrade to find out :)
